x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/user/ChangePasswordRequest.java

```diff
@@ -92,4 +95,21 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeBytesReference(new BytesArray(CharArrays.toUtf8Bytes(passwordHash)));
         refreshPolicy.writeTo(out);
     }
+
+    @Override
+    public final XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject()
+                .startObject("change_password")
+                .field("username", username);
+        if (params.paramAsBoolean(AuditToXContentParams.INCLUDE_CREDENTIALS, false)) {
+            builder.field("password_hash", passwordHash != null ? String.valueOf(passwordHash) : null);
+        } else {
+            builder.field("password_hash", passwordHash != null ? "<redacted>" : null);
+        }
+        builder.endObject(); // change_password
+        if (params.paramAsBoolean(AuditToXContentParams.INCLUDE_REFRESH_POLICY, false)) {
```

Comment:
Yes, verbosity compared to the usefulness of the information.
Because the event is recorded before the config change actually takes effect, a refresh policy of `immediate` only says that the change will take place _sooner_ after the the event is recorded, compared to the other options (i.e. `none` and `wait_until`).

However, the refresh value is still optionally included, because the `toXContent` method is generic and hence should cover all the request fields.